### PR TITLE
disable public interfaces & pool based cloud init

### PR DESCRIPTION
**Pull request by derlinuxer**
this PR will adds two configuration features

- add config parameter to disable public interfaces at hetzner node creation (ipv4 and/or ipv6)
- cloud init function can be configured in pool section too (pool config overrules global config)

thanks and take your time no harry to this

stephan

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -214,7 +218,9 @@ hetzner-k3s create --config cluster_config.yaml
 
 This will take a few minutes depending on the number of masters and worker nodes.
 
-
+### disable public IPs (IPv4 or IPv6 or both) on nodes
+With `enable_public_net_ipv4: false` and `enable_public_net_ipv6: false` you can disable public interface at hetzner node creation. This settings are global and effects all master and worker. If you disable public IPs be sure that your hetzer-k3s executor has excess to the hetzner private network the nodes will be deployed in (executor is already in the privNet or has VPN excess).
```
Comment (jrudolph) [inline README.md:222]:
```suggestion
With `enable_public_net_ipv4: false` and `enable_public_net_ipv6: false` you can disable public interface at hetzner node creation. This settings are global and effects all master and worker. If you disable public IPs be sure that your hetzer-k3s executor has excess to the hetzner private network the nodes will be deployed in (executor is already in the privNet or has VPN access).
```

**Comment by vitobotta** [comment]
@derlinuxer hi, why did you close it? :)

**Comment by derlinuxer** [comment]
I tried to sync your changes into my fork via "sync fork" ("sync fork" should be called "override fork"). That was a mistake and all my changes were gone, also the PR. We use your tool very much and I needed your fix for "system_upgrade_controller_manifest_url", but me changes are also mandatory. Now I merged it with my local git from IDE and will create a new RP after some tests.
Do you have any advice for a proper sync your changes into my fork if the PR not included and without loosing my stuff?

**Comment by vitobotta** [comment]
Oh sorry to hear that. I would usually just pull from the origin repo/branch to update mine.

**Pull request closed without merging**